src/component/feature/FeatureView/FeatureVariants/FeatureVariantsList/AddFeatureVariant/AddFeatureVariant.tsx

```diff
@@ -67,15 +62,17 @@ const AddVariant = ({
     const { projectId, featureId } = useParams<IFeatureViewParams>();
     const { feature } = useFeature(projectId, featureId);
     const [variants, setVariants] = useState<IFeatureVariant[]>([]);
-    const [code, setCode] = useState('');
 
-    const isJSON = () => {
+    const isJSON = (input: string): boolean => {
         try {
-            let parsedString = JSON.parse(code);
-            if (parsedString && typeof parsedString === 'object') {
-                return true;
-            }
-        } catch (e) {
+            JSON.parse(input);
+            return true;
+        } catch (e: unknown) {
+            setError({
+                payload: `The payload is not a valid JSON format: ${e
```

Comment:
Maybe `'Invalid JSON.'` would be enough (if this is supposed to be used for small JSON snippets).